examples/system/cpp_pthread/main/cpp_pthread.cpp

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <memory>
+#include <string>
+#include <sstream>
+#include <esp_pthread.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <esp_log.h>
+
+using namespace std::chrono;
+
+const auto sleep_time = seconds{5};
+
+void print_thread_info(const char* extra = nullptr)
+{
+        std::stringstream ss;
+        if(extra)
+        {
+            ss << extra;
+        }
+        ss << "Core id: " << xPortGetCoreID()
+            << ", prio: " << uxTaskPriorityGet(nullptr) 
+            << ", minumim free stack: " << uxTaskGetStackHighWaterMark(nullptr) << " bytes.";
+        ESP_LOGI(pcTaskGetTaskName(nullptr), "%s", ss.str().c_str());        
```

Comment:
Wouldn't it be nice if we could keep using Pthread APIs here... Oh well, maybe some other time.

Reply:
It would, but I guess it's more to it than just wrapping the functions?

Reply:
Yes, let's keep this for now.